### PR TITLE
Rollback full release-line

### PR DESCRIPTION
**Pull request by msluther**
This enables a full release-line to be rolled back if there are dependent packages.

There's some sketchiness in the rollback of dependent packages if some of them were to fail for some reason. I've tried to leave comments in place where those exist.  I'm not sure there's a real good full-featured solution though since the whole thing is asynchronous and potentially out of process/on another server if a full build is ever needed.

### CODE REVIEW

File: lib/npm/routes.js [hunk 1]
```diff
@@ -223,14 +245,65 @@ module.exports = function (app) {
...
+        earlyReturn: false
+      }));
+
+    const config = {
+      retry: { retries: 5, min: 50, max: 10000 },
+      limit: 5,
+      ...app.config.get('dependentBuilds')
+    };
+
+    // setImmediate(function () {
```
Comment (fritzmonkey) [inline lib/npm/routes.js]: This can be removed.

File: test/integration/npm/routes.rollback.test.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+  after(function (done) {
+    async.series([
+      ...cleanupWork,
+      app.close.bind(app),
+      registry.close.bind(registry)
+    ], done);
+  });
+
+  // TODO: undo all that work in `after`
```
Comment (fritzmonkey) [inline test/integration/npm/routes.rollback.test.js]: Is this TODONE above?

File: lib/npm/routes.js (hunk 1 above)
Comment (msluther) [reply lib/npm/routes.js]: *sigh* yeah, thought I got all this stuff :(

File: test/integration/npm/routes.rollback.test.js (hunk 2 above)
Comment (msluther) [reply test/integration/npm/routes.rollback.test.js]: lol yeah. this was a note to myself as I was on my way home. 😂

**Pull request merged**